cmd/virt-launcher/virt-launcher.go

```diff
@@ -610,16 +617,28 @@ func RemoveContents(dir string) error {
 
 func terminateIstioProxy() {
 	if istioProxyPresent() {
-		resp, err := http.Post(fmt.Sprintf("http://localhost:%d/quitquitquit", infraconfigurators.EnvoyMergedPrometheusTelemetryPort), "", nil)
-		if err != nil || resp.StatusCode != http.StatusOK {
-			log.Log.Error("Failed to request Istio proxy termination")
+		isRetriable := func(err error) bool {
+			if net.IsConnectionReset(err) || net.IsConnectionRefused(err) || errors2.IsServiceUnavailable(err) {
+				return true
+			}
+			return false
 		}
+		err := retry.OnError(retry.DefaultBackoff, isRetriable, func() error {
+			resp, err := httpClient.Post(fmt.Sprintf("http://localhost:%d/quitquitquit", infraconfigurators.EnvoyMergedPrometheusTelemetryPort), "", nil)
+			if err != nil || resp.StatusCode != http.StatusOK {
+				log.Log.Error("failed to request istio-proxy termination, retrying...")
```

Comment:
It'd be useful to have the `err` or the `StatusCode` printed alongside the error message.

Reply:
Added